### PR TITLE
[13.0][ADD] sale_coupon_commercial_partner_applicability: New module

### PR DESCRIPTION
**Pull request by chienandalu**
cc @Tecnativa TT41522

please review @CarlosRoca13 @sergio-teruel

### CODE REVIEW

File: sale_coupon_apply_commercial_partner/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "summary": "Coupons can be used by all the commercial entity partners",
+    "version": "13.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Sale",
+    "website": "https://github.com/OCA/sale-promotion",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "maintainers": ["chienandalu"],
+    "license": "AGPL-3",
+    "depends": ["sale_coupon"],
+    "data": [],
```
Comment (sergio-teruel) [inline sale_coupon_apply_commercial_partner/__manifest__.py]: Remove this key

File: sale_coupon_apply_commercial_partner/models/coupon_coupon.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    _inherit = "sale.coupon"
+
+    def _check_coupon_code(self, order):
+        message = super()._check_coupon_code(order)
+        if message.get("error", "") == _("Invalid partner."):
+            if (
+                self.partner_id.commercial_partner_id
+                == order.partner_id.commercial_partner_id
+            ):
+                message.pop("error")
```
Comment (sergio-teruel) [inline sale_coupon_apply_commercial_partner/models/coupon_coupon.py]:
```suggestion
                message.pop("error", None)
```

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Review by CarlosRoca13 — APPROVED** [review]
LGTM but please note the previous comments

File: sale_coupon_apply_commercial_partner/models/coupon_coupon.py (hunk 2 above)
Comment (chienandalu) [reply sale_coupon_apply_commercial_partner/models/coupon_coupon.py]: Not necessary, as the previous `if` makes sure we've got the key.

**Comment by pedrobaeza** [comment]
The name should be reversed: `sale_coupon_commercial_partner_apply` or `sale_coupon_commercial_partner_applicability`.

**Comment by chienandalu** [comment]
Changed

**Comment by CarlosRoca13** [comment]
This PR can be merged?

**Pull request merged**